zcash_client_sqlite/src/wallet.rs

```diff
@@ -3054,6 +3063,67 @@ pub(crate) fn set_transaction_status(
     Ok(())
 }
 
+pub(crate) fn refresh_status_requests(
+    conn: &rusqlite::Transaction,
+) -> Result<(), SqliteClientError> {
+    if let Some(chain_tip) = chain_tip_height(conn)? {
+        let mut unmined_query = conn.prepare(
+            "SELECT
+                t.txid,
+                t.expiry_height,
+                r.query_type IS NOT NULL AS status_request_exists
+             FROM transactions t
+             LEFT OUTER JOIN tx_retrieval_queue r ON r.txid = t.txid
+             WHERE t.mined_height IS NULL
+             AND (
+                t.expiry_height IS NULL -- expiry is unknown
+                OR t.expiry_height = 0 -- tx will note expire
+                OR t.expiry_height > :chain_tip -- tx is unexpired
+             )
+             AND (
+                r.query_type IS NULL
+                OR r.query_type = :status_type
+             )",
+        )?;
+
+        let mut rows = unmined_query.query(named_params! {
+            ":chain_tip": u32::from(chain_tip),
+            ":status_type": TxQueryType::Status.code(),
+        })?;
+        while let Some(row) = rows.next()? {
+            let txid = row.get::<_, Vec<u8>>("txid")?;
+            let tx_expiry = row.get::<_, Option<u32>>("expiry_height")?;
+
+            if row.get::<_, bool>("status_request_exists")? {
+                conn.execute(
+                    "UPDATE tx_retrieval_queue
+                     SET request_expiry = :new_expiry
+                     WHERE txid = :txid
+                     AND query_type = :status_type",
+                    named_params! {
+                        ":new_expiry": tx_retrieval_expiry(chain_tip, tx_expiry),
+                        ":txid": &txid[..],
+                        ":status_type": TxQueryType::Status.code(),
+                    },
+                )?;
+            } else {
+                conn.execute(
+                    "INSERT INTO tx_retrieval_queue (txid, query_type, request_expiry)
+                     VALUES (:txid, :status_type, :new_expiry)
+                     ON CONFLICT (txid) DO NOTHING",
```

Comment:
This could have been `ON CONFLICT (txid) DO UPDATE SET request_expiry = IIF(query_type = :status_type, :new_expiry, request_expiry)`. And then `status_request_exists` is not needed and the `INSERT ... ON CONFLICT ...` can be unconditional.

(`ON CONFLICT (txid) DO UPDATE SET request_expiry = IIF(query_type = :status_type, :new_expiry, request_expiry)` would more closely match the existing logic, but as noted above the `AND query_type = :status_type` condition is unnecessary.)